src/monodroid/jni/xamarin_getifaddrs.cc

```diff
@@ -667,34 +676,41 @@ fill_sa_address (struct sockaddr **sa, struct ifaddrmsg *net_address, void *rta_
 }
 
 static int
-fill_ll_address (struct sockaddr_ll_extended **sa, struct ifinfomsg *net_interface, void *rta_data, int rta_payload_length)
+fill_ll_address (struct sockaddr_ll_extended **sa, struct ifinfomsg *net_interface, void *rta_data, size_t rta_payload_length)
 {
 	assert (sa);
 	assert (net_interface);
-	
+
 	/* Always allocate, do not free - caller may reuse the same variable */
-	*sa = new sockaddr_ll_extended (); //calloc (1, sizeof (**sa));
```

Comment:
Why *not* use `new sockaddr_ll_extended()`?  Is `calloc()` better for some reason now?

Reply:
This file is a potential candidate for future move to the Mono runtime and, as such, it's best kept as `C` as possible - thus the change back.